pagopa/io-app-design-system
# chore: Add `ModuleNavigation` component

## Short description
This PR adds the `ModuleNavigation` component

## List of changes proposed in this pull request
- Added `ModuleNavigation` component
- Added examples in the example app

## How to test
Navigate to the Module screen and check that the component is correctly rendered

## Preview
<img width="250" alt="Screenshot 2024-04-30 alle 11 06 02" src="https://github.com/pagopa/io-app-design-system/assets/6160324/da252a8f-5c88-4f30-890e-ea4ce9b78b39">
